### ISSUE TITLE
[Example] Request: Add https -> http redirect example

### ISSUE THREAD

**Issue by mcjcloud** [hatoo/http-mitm-proxy#20]
The `dev_proxy.rs` example shows a redirect to a local server, but it only works with `dev.example`. If you replace that host with an https host (e.g. youtube.com), the request either hangs indefinitely or returns a 403/400.

Could you add an example, similar to `dev_proxy.rs`, that redirects https requests to a local http server?

**Comment by mcjcloud** [comment]
With some further investigation, I was able to get it to work by simply adding
```rust
parts.path_and_query = match parts.path_and_query {
    None => Some(PathAndQuery::from_static("")),
    _ => parts.path_and_query,
};
```

**Comment by mcjcloud** [comment]
Another update. This fix works in the repo, but not in the most recent release (0.6.1).

Using ```rust
[dependencies]
http-mitm-proxy = "0.6.1"
```
does not work, but using master branch
```rust
[dependencies]
http-mitm-proxy = { path = "../../../hatoo/http-mitm-proxy" }
```
does.

**Comment by hatoo** [comment]
Hi, thanks for your issue!

I think it is fixed by #21 

I'll publish next release soon
